File: 2 Pointers/Squares of a Sorted Array/Solution.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        vector<int> result;
        result.reserve(nums.size());
        for (size_t i = 0; i < nums.size(); ++i) {
            const int square = nums[i] * nums[i];
            result.push_back(square);
        }

        sort(result.begin(), result.end());
        return result;
    }
};

File: 2 Pointers/Squares of a Sorted Array/Solution.kt
harlclass Solution {
    fun sortedSquares(nums: IntArray): IntArray {
        var left = 0
        var right = nums.size - 1
        val result = IntArray(nums.size)
        var index = nums.size - 1

        while (left <= right) {
            val leftSquare = nums[left] * nums[left]
            val rightSquare = nums[right] * nums[right]
            if (leftSquare > rightSquare) {
                result[index--] = leftSquare
                left++
            } else {
                result[index--] = rightSquare
                right--
            }
        }
        return result
    }
}